samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -133,4 +139,12 @@ final void writePlanJsonFile(String planJson) {
       log.warn("Failed to write execution plan json to file", e);
     }
   }
+
+  private void validateConfig() {
+    String appClassName = new ApplicationConfig(config).getAppClass();
+    if (appClassName != null && !config.containsKey(TaskConfig.WINDOW_MS())) {
```

Comment:
Currently both window and join require task.window.ms.

We could do traversals to figure out if we have those operators, and we could also override the default value of task.window.ms to something other than -1 based on heuristics, but I'd prefer to keep this simple and explainable for now. WDYT?